pc/storage/kdtree/kdtree_test.go

```diff
@@ -54,36 +54,117 @@ func TestKDtree(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	return it
+}
+
+func TestKDtree(t *testing.T) {
+	it := createTestPointCloud(t)
 	kdt := New(it)
 
-	expectedTree := &KDTree{
-		Vec3RandomAccessor: it,
-		root: &node{
-			children: [2]*node{
-				&node{
+	t.Run("Equal", func(t *testing.T) {
```

Comment:
I don't think this test validates `KDTree.Equal()`'s algorithm.
Test of Equal would be something like: https://github.com/seqsense/pcgol/commit/c46372e1c8c3f43bde490561e975be48b59fee08

(As it is not very easy to validate such recursive algorithm, I used `reflect.DeepEqual()` in the tests)

Reply:
Do you suggest to initialize `poolNodeArray` in `tt.expectedTree`?

Reply:
Maybe we can use that function to other tests to keep them independent from `Equal`.